### PR TITLE
fix: replace tags with digests in external_images.txt

### PR DESCRIPTION
**Pull request by svor**
Signed-off-by: svor <vsvydenk@redhat.com>

### What does this PR do?
Replaces tags with digests in external_images.txt for both registries.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-2270

### CODE REVIEW

File: dependencies/che-plugin-registry/build/dockerfiles/entrypoint.sh [hunk 1]
```diff
@@ -187,7 +204,7 @@ function update_container_image_references() {
     # We can't use the `-d` option for readarray because
     # registry.centos.org/centos/httpd-24-centos7 ships with Bash 4.2
     # The below command will fail if any path contains whitespace
-    readarray -t metas < <(find "${METAS_DIR}" -name 'meta.yaml' -o -name 'devfile.yaml')
+    readarray -t metas < <(find "${METAS_DIR}" -name 'meta.yaml' -o -name 'devfile.yaml' -o -name 'che-theia-plugin.yaml')
```
Comment (nickboldt) [inline dependencies/che-plugin-registry/build/dockerfiles/entrypoint.sh:207]: any other yamls we need to change here too?

File: dependencies/che-plugin-registry/build/dockerfiles/test_entrypoint.sh [hunk 2]
```diff
@@ -776,3 +776,31 @@ source "${script_dir}/entrypoint.sh"
 extract_and_use_related_images_env_variables_with_image_digest_info
 
 assertFileContentEquals "${METAS_DIR}/che-theia-plugin.yaml" "${expected_cheTheiaPluginYaml}"
+
+#################################################################
+initTest "Should replace image references in external_images.txt with RELATED_IMAGE env vars"
```
Comment (nickboldt) [inline dependencies/che-plugin-registry/build/dockerfiles/test_entrypoint.sh:781]: should we have a test_entrypoint.sh for devfile reg too?

**Review by nickboldt — APPROVED** [review]
LGTM, with 2 questions

File: dependencies/che-plugin-registry/build/dockerfiles/entrypoint.sh (hunk 1 above)
Comment (nickboldt) [reply dependencies/che-plugin-registry/build/dockerfiles/entrypoint.sh:207]:
```
$➔ grep -r :2.1
build/dockerfiles/test_entrypoint.sh:      image: 'registry.redhat.io/codeready-workspaces/theia-rhel8:2.11'
...
build/dockerfiles/test_entrypoint.sh:  image: 'registry.redhat.io/codeready-workspaces/plugin-java11-rhel8:2.11'
che-editors.yaml:          image: 'registry.redhat.io/codeready-workspaces/theia-rhel8:2.12'
...
che-editors.yaml:          image: 'registry.redhat.io/codeready-workspaces/theia-endpoint-rhel8:2.12'
che-theia-plugins.yaml:      image: "registry.redhat.io/codeready-workspaces/plugin-java8-rhel8:2.12"
...
che-theia-plugins.yaml:      image: "registry.redhat.io/codeready-workspaces/plugin-java8-rhel8:2.12"
che-plugins.yaml:        image: 'registry.redhat.io/codeready-workspaces/machineexec-rhel8:2.12'
```

Comment (svor) [reply dependencies/che-plugin-registry/build/dockerfiles/entrypoint.sh:207]: I don't see other files, seems it's only meta.yaml, devfile.yaml and che-theia-plugin.yaml

Comment (svor) [reply dependencies/che-plugin-registry/build/dockerfiles/entrypoint.sh:207]: che-editors.yaml and che-theia-plugins.yaml will be transformed into meta.yamls and the script will change all meta.yaml files at runtime. So, everything should be ok.

File: dependencies/che-plugin-registry/build/dockerfiles/test_entrypoint.sh (hunk 2 above)
Comment (svor) [reply dependencies/che-plugin-registry/build/dockerfiles/test_entrypoint.sh:781]: now we have it

File: dependencies/che-plugin-registry/build/dockerfiles/entrypoint.sh (hunk 1 above)
Comment (nickboldt) [reply dependencies/che-plugin-registry/build/dockerfiles/entrypoint.sh:207]: so where does `che-theia-plugin.yaml` come from? I only found a `che-plugins.yaml` and a `che-theia-plugins.yaml` ? is it generated too, like meta.yaml and devfile.yaml?

Comment (svor) [reply dependencies/che-plugin-registry/build/dockerfiles/entrypoint.sh:207]: yes, `che-theia-plugin.yaml` (no `s`) is generated from che-theia-plugins.yaml (with `s`).

**Comment by nickboldt** [comment]
cherrypicked to 2.12  as 820e0b6f

**Pull request merged**